### PR TITLE
Added enum values for serialization / deserialization

### PR DESCRIPTION
**Pull request by Kun-Kun**

### CODE REVIEW

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/ConversationType.java [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+import java.util.Map;
+
 /**
  * Created by user on 26.03.2017.
  */
 public enum ConversationType {
-    INDIVIDUAL,GROUP;
+    @JsonProperty("0")
+    INDIVIDUAL,
+    @JsonProperty("1")
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/ConversationType.java]: @Kun-Kun No need to add bunch of this JsonProperty annotations. Make enum with constructor and use annotations JsonCreator and JsonValue

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/MessageStatus.java [hunk 2]
```diff
@@ -1,8 +1,19 @@
 package com.softgroup.messenger.api.dto.enumeration;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 /**
  * Created by user on 26.03.2017.
  */
 public enum MessageStatus {
-    GENERATED,SENT,DELIVERED,VIEWED,DELETED
+    @JsonProperty("0")
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/MessageStatus.java]: @Kun-Kun same

File: softgroup/basis/common-dao/common-dao-api/pom.xml [hunk 3]
```diff
@@ -11,6 +11,13 @@
 
     <artifactId>common-dao-api</artifactId>
     <packaging>jar</packaging>
+    <dependencies>
+        <dependency>
+            <groupId>com.softgroup</groupId>
+            <artifactId>messenger-api</artifactId>
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-api/pom.xml]: @Kun-Kun DAO should not depend on messenger

File: softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/ConversationType.java [hunk 4]
```diff
@@ -1,16 +1,45 @@
...
-    INDIVIDUAL,
-    @JsonProperty("1")
-    GROUP;
+
+    INDIVIDUAL(0),
+    GROUP(1);
+
+    private Integer value;
+
+    private static final Map<Integer,ConversationType> TYPE_MAP ;
```
Comment (Dominys) [inline softgroup/services/messenger/messenger-api/src/main/java/com/softgroup/messenger/api/dto/enumeration/ConversationType.java]: @Kun-Kun Why you need this ??

File: softgroup/basis/common-dao/common-dao-api/pom.xml (hunk 3 above)
Comment (Kun-Kun) [inline softgroup/basis/common-dao/common-dao-api/pom.xml]: @Dominys Dao use same enum like DTO

**Pull request merged**